src/nvim/eval.c

```diff
@@ -12867,13 +12867,17 @@ static void f_rpcstart(typval_T *argvars, typval_T *rettv)
   char **argv = xmalloc(sizeof(char_u *) * argvl);
 
   // Copy program name
+  if (argvars[0].vval.v_string == NULL || argvars[0].vval.v_string[0] == NUL) {
+    EMSG(_(e_api_spawn_failed));
+    return
+  }
   argv[0] = xstrdup((char *)argvars[0].vval.v_string);
 
   int i = 1;
   // Copy arguments to the vector
   if (argsl > 0) {
     for (listitem_T *arg = args->lv_first; arg != NULL; arg = arg->li_next) {
-      argv[i++] = xstrdup((char *)arg->li_tv.vval.v_string);
+      argv[i++] = xstrdup((char *) get_tv_string(&arg->li_tv));
```

Comment:
Same problem in `jobstart()`, where we assume that argument strings never point to `NULL`.